torch/_library/opaque_object.py

```diff
@@ -1,10 +1,36 @@
 from typing import Any, NewType
 
 import torch
+import torch.utils._pytree as pytree
 
 from .fake_class_registry import register_fake_class
 
 
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Note [Opaque Objects]
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# Opaque objects are objects that we allow to be passed into custom operators.
```

Comment:
"We allow users to specify that custom operators may accept their own user-defined opaque objects as inputs"